File: sysadmin/docker/Dockerfile.bitsy.rs
FROM ubuntu:22.04 AS builder

WORKDIR /app

COPY . .

RUN apt update && apt install -y \
    build-essential \
    curl \
    libpq-dev \
    software-properties-common \
    curl

RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y

ENV PATH="/root/.cargo/bin:${PATH}"

RUN cargo build --release --all-targets


FROM ubuntu:22.04

RUN apt-get update -y \
    && apt-get install -y --no-install-recommends \
    ca-certificates \
    libpq-dev \
    # Clean up
    && apt-get autoremove -y \
    && apt-get clean -y

COPY --from=builder /app/target/release/bitsy .
COPY --from=builder /app/target/release/bitsy.d .


EXPOSE ${PORT}

RUN chmod +x ./bitsy
CMD exec ./bitsy --host "${HOST}" --port "${PORT}" --pg-host "${PG_HOST}" --pg-port "${PG_PORT}" --pg-password "${PG_PASSWORD}"

File: sysadmin/docker/Dockerfile.bitsy.py
FROM ubuntu:22.04

WORKDIR /app

ENV DEBIAN_FRONTEND noninteractive

RUN apt update && apt-get upgrade -y
RUN apt install software-properties-common -y
RUN add-apt-repository ppa:deadsnakes/ppa -y
RUN apt install -y \
    libpq-dev \
    build-essential \
    python3.9 \
    python3.9-dev \
    python3.9-venv \
    python3.9-distutils \
    python3.9-lib2to3 \
    python3.9-gdbm

COPY . .

RUN python3.9 -m venv venv
RUN ./venv/bin/pip install -r requirements/requirements.txt

EXPOSE 8000

CMD exec ./venv/bin/python3.9 main.py --api-host "${API_HOST}" --api-port "${API_PORT}" --pg-host "${PG_HOST}" --pg-port "${PG_PORT}" --pg-user "${PG_USER}" --pg-password "${PG_PASSWORD}"

# docker run -it -p 8000:8000 -e "PG_HOST=6.tcp.ngrok.io" -e "PG_PORT=19705" -e "PG_USER=postgres" -e "pg_database=bitsy" ralston3/bitsy:py-latest
